gamemodes/terrortown/gamemode/client/cl_search.lua

```diff
@@ -2,723 +2,393 @@
+		}, 62)
+	elseif GetConVar("ttt2_inspect_confirm_mode"):GetInt() == 0 and not bodysearch.IsConfirmed(data.ragOwner) then
+		-- a detective can only be called AFTER a body was confirmed
+		self:MakeInfoItem(contentAreaScroll, "policingrole_call_confirm", {
+			text = {
+				title = {
+					body = "search_title_policingrole_report_confirm",
+					params = nil
+				},
+				text = {{
+					body = "search_policingrole_report_confirm",
+					params = nil
+				}}
+			},
+			colorBox = roles.DETECTIVE.ltcolor
+		}, 62)
+	elseif bodysearch.CanReportBody(data.ragOwner)
+		and not bodysearch.IsConfirmed(data.ragOwner)
+		and not (clientRD.isPolicingRole and clientRD.isPublicRole)
+	then
+		local searchMode = bodysearch.GetInspectConfirmMode()
+
+		self:MakeInfoItem(contentAreaScroll, "policingrole_confirm_disabled", {
+			text = {
+				title = {
+					body = "search_title_policingrole_confirm_disabled",
+					params = nil
+				},
+				text = {{
+					body = "search_policingrole_confirm_disabled_" .. tostring(searchMode),
+					params = nil
+				}}
+			},
+			colorBox = roles.DETECTIVE.ltcolor
+		}, (searchMode == 1) and 62 or 78)
 	end
 
-	-- Finalize search data, prune stuff that won't be shown etc
-	-- search is a table of tables that have an img and text key
-	local search = PreprocSearch(search_raw)
-
-	-- Install info controller that will link up the icons to the text etc
-	dlist.OnActivePanelChanged = SearchInfoController(search, dactive, dtext)
-
-	-- Create table of SimpleIcons, each standing for a piece of search
-	-- information.
-	local start_icon
-
-	for t, info in SortedPairsByMemberValue(search, "p") do
-		local ic
-		local icon = info.img
-
-		-- Certain items need a special icon conveying additional information
-		if t == "nick" then
-			local avply = IsValid(search_raw.owner) and search_raw.owner or nil
+		-- POPULATE WITH INFORMATION
+		for i = 1, #bodysearch.searchResultOrder do
+			local searchResultName = bodysearch.searchResultOrder[i]
+			local searchResultData = bodysearch.GetContentFromData(searchResultName, data)
 
-			ic = vgui.Create("SimpleIconAvatar", dlist)
-			ic:SetPlayer(avply)
+			if not searchResultData then continue end
 
-			start_icon = ic
-		elseif t == "lastid" then
-			ic = vgui.Create("SimpleIconAvatar", dlist)
-			ic:SetPlayer(info.ply)
-			ic:SetAvatarSize(24)
-		elseif info.text_icon then
-			ic = vgui.Create("SimpleIconLabelled", dlist)
-			ic:SetIconText(info.text_icon)
-		elseif t == "role" then
-			ic = vgui.Create("SimpleRoleIcon", dlist)
-
-			ic.Icon:SetImage2("vgui/ttt/dynamic/icon_base_base")
-			ic.Icon:SetImageOverlay("vgui/ttt/dynamic/icon_base_base_overlay")
-			ic.Icon:SetRoleIconImage(icon)
-
-			icon = "vgui/ttt/dynamic/icon_base"
-		else
-			ic = vgui.Create("SimpleIcon", dlist)
+			self:MakeInfoItem(contentAreaScroll, searchResultName, searchResultData)
 		end
 
-		ic:SetIconSize(64)
-		ic:SetIcon(icon)
-
-		if info.color then
-			ic:SetIconColor(info.color)
+	if roleKnown then
```

Comment:
I don't understand why you're trying to say here

Reply:
You only add other addons content if the roleKnown variable is true, is there a reason for this? Eg. Is this the same as it is currently?

Reply:
currently there is no option for the role being unknown as this is a new mode (that was many times requested on the discord).
But since there is no info besides the playermodel shown, when the role is not known, I don't want to show any additional info either